lib/Wikia/src/Util/SamplerProxy.php

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * A wrapper for a POPO that supports dispatching a specific method to an alternate
+ * object, either as a shadowed execution (i.e. execute both the original method and the
+ * alternate) or as a substitution (i.e. execute the alternate instead of the original).
+ * Both shadowing and substitution may be sampled so that only a specified percentage of
+ * calls involve the alternate and the rest pass through to the original.
+ */
+namespace Wikia\Util;
+
+
+class SamplerProxy {
+
+	/**
+	 * @var string
+	 */
+	protected $enableShadowingVariableName;
+
+	/**
+	 * @var string
+	 */
+	protected $methodSamplingRateVariableName;
+
+	/**
+	 * @var callable
+	 * Expected to be of the form [ <instance>, <method name> ]
+	 */
+	protected $originalCallable;
+
+	/**
+	 * @var callable
+	 * Expected to be of the form [ <instance>, <method name> ]
+	 * Should have the same argument list as $originalCallable
+	 */
+	protected $alternateCallable;
+
+	/**
+	 * @var callable
+	 * Takes 2 arguments, ( $originalResults, $alternateResults ) and
+	 * returns the results that should be passed back to the caller
+	 */
+	protected $resultsCallable;
+
+	public static function createBuilder() {
+		return new SamplerProxyBuilder();
+	}
+
+	public function __construct( SamplerProxyBuilder $samplerProxyBuilder ) {
+		$this->enableShadowingVariableName = $samplerProxyBuilder->getEnableShadowingVariableName();
+		$this->methodSamplingRateVariableName =
+			$samplerProxyBuilder->getMethodSamplingRateVariableName();
+		$this->originalCallable = $samplerProxyBuilder->getOriginalCallable();
+		$this->alternateCallable = $samplerProxyBuilder->getAlternateCallable();
+		$this->resultsCallable = $samplerProxyBuilder->getResultsCallable();
+	}
+
+	public function __call( $method, $args ) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce the number of returns of this function 5, down to the maximum allowed 3. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1142)